### PR TITLE
JBIDE-27465: Added ApplicationYaml test

### PR DESCRIPTION
**Pull request by olkornii**
Signed-off-by: olkornii <olkornii@redhat.com>

# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work? https://studio-jenkins-csb-codeready.cloud.paas.psi.redhat.com/job/Studio/job/Quality/job/Integration-tests/job/quarkus-itests/246/

### CODE REVIEW

File: itests/org.jboss.tools.quarkus.integration.tests/src/org/jboss/tools/quarkus/integration/tests/application/yaml/ApplicationYaml.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+		addExtensionInPomFile(editor);
+
+		createNewFile(PROJECT_NAME, APPLICATION_YAML_NAME, APPLICATION_YAML_PATH);
+
+		addLinesInApplicationYaml(editor);
+
+		WorkbenchShellHandler.getInstance().closeAllNonWorbenchShells();
+
+		new QuarkusLaunchConfigurationTabGroup().selectProject(PROJECT_NAME);
+		new QuarkusLaunchConfigurationTabGroup().openRunConfiguration();
```
[the remark this answers is not in the document]
Comment (olkornii) [reply itests/org.jboss.tools.quarkus.integration.tests/src/org/jboss/tools/quarkus/integration/tests/application/yaml/ApplicationYaml.java:78]: fixed

File: itests/org.jboss.tools.quarkus.integration.tests/src/org/jboss/tools/quarkus/integration/tests/project/universal/methods/AbstractQuarkusTest.java [hunk 2]
```diff
@@ -110,29 +107,17 @@ public static void checkProblemsView() {
...
-		URL localhost = new URL(readedURL);
-		String inputLine = "";
-		try (BufferedReader in = new BufferedReader(new InputStreamReader(localhost.openStream()));) {
-			inputLine = in.readLine();
-			in.close();
-		} catch (IOException e) {
-			QuarkusCorePlugin.logException("Can`t read from url!", e);
-		}
-		return inputLine;
+		new ProjectExplorer().getProject(projectName).getProjectItem(filePath).select();
```
[the remark this answers is not in the document]
Comment (olkornii) [reply itests/org.jboss.tools.quarkus.integration.tests/src/org/jboss/tools/quarkus/integration/tests/project/universal/methods/AbstractQuarkusTest.java]: fixed

File: itests/org.jboss.tools.quarkus.integration.tests/src/org/jboss/tools/quarkus/integration/tests/project/universal/methods/AbstractQuarkusTest.java [hunk 3]
```diff
@@ -110,29 +107,17 @@ public static void checkProblemsView() {
...
-		} catch (IOException e) {
-			QuarkusCorePlugin.logException("Can`t read from url!", e);
-		}
-		return inputLine;
+	public static void createNewFile(String projectName, String fileName, String filePath) {
+		new WorkbenchShell().setFocus();
+		ProjectExplorer pe = new ProjectExplorer();
+		pe.selectProjects(projectName);
+		pe.getProject(projectName).getProjectItem(filePath).select();
+		new ContextMenuItem(TextLabels.NEW_CONTEXT_ITEM, "Other...").select();
```
Comment (jkopriva) [inline itests/org.jboss.tools.quarkus.integration.tests/src/org/jboss/tools/quarkus/integration/tests/project/universal/methods/AbstractQuarkusTest.java:115]: I think you can use this: https://github.com/eclipse/reddeer/blob/37f27affd709a9622d84e8d29d047dc78a0de1b4/plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/selectionwizard/NewMenuWizard.java

File: itests/org.jboss.tools.quarkus.integration.tests/src/org/jboss/tools/quarkus/integration/tests/project/universal/methods/AbstractQuarkusTest.java [hunk 4]
```diff
@@ -110,29 +107,17 @@ public static void checkProblemsView() {
...
+		new LabeledText("Wizards:").setText("File");
+		new NextButton().click();
+		new LabeledText("File name:").setText(fileName);
```
Comment (jkopriva) [inline itests/org.jboss.tools.quarkus.integration.tests/src/org/jboss/tools/quarkus/integration/tests/project/universal/methods/AbstractQuarkusTest.java:118]: and also this: https://github.com/eclipse/reddeer/blob/37f27affd709a9622d84e8d29d047dc78a0de1b4/plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/ui/dialogs/WizardNewFileCreationPage.java

File: itests/org.jboss.tools.quarkus.integration.tests/src/org/jboss/tools/quarkus/integration/tests/project/universal/methods/AbstractQuarkusTest.java (hunk 3 above)
Comment (olkornii) [reply itests/org.jboss.tools.quarkus.integration.tests/src/org/jboss/tools/quarkus/integration/tests/project/universal/methods/AbstractQuarkusTest.java]: reworked

File: itests/org.jboss.tools.quarkus.integration.tests/src/org/jboss/tools/quarkus/integration/tests/project/universal/methods/AbstractQuarkusTest.java (hunk 4 above)
Comment (olkornii) [reply itests/org.jboss.tools.quarkus.integration.tests/src/org/jboss/tools/quarkus/integration/tests/project/universal/methods/AbstractQuarkusTest.java]: reworked

File: itests/org.jboss.tools.quarkus.integration.tests/src/org/jboss/tools/quarkus/integration/tests/application/yaml/ApplicationYaml.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jkopriva) [reply itests/org.jboss.tools.quarkus.integration.tests/src/org/jboss/tools/quarkus/integration/tests/application/yaml/ApplicationYaml.java:78]: Maybe this needs more reworking, see https://github.com/eclipse/reddeer/blob/master/plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/debug/ui/launchConfigurations/LaunchConfigurationsDialog.java

**Review by jkopriva — CHANGES_REQUESTED** [review]

**Comment by jkopriva** [comment]
@olkornii please review all code smells.

File: itests/org.jboss.tools.quarkus.integration.tests/src/org/jboss/tools/quarkus/integration/tests/project/RunProjectWithDebugTest.java [hunk 5]
```diff
@@ -79,13 +79,12 @@ public void testRunWithDebug() {
...
-		new QuarkusLaunchConfigurationTabGroup().openDebugConfiguration();
+		DebugConfigurationsDialog debugDialog = new DebugConfigurationsDialog();
+		QuarkusLaunchConfigurationTabGroup launchConfiguration = new QuarkusLaunchConfigurationTabGroup();
 
-		new DefaultTreeItem(TextLabels.QUARKUS_APPLICATION_TREE_ITEM).select();
-		new ContextMenuItem("New Configuration").select();
-
-		new PushButton(TextLabels.DEBUG).click();
+		debugDialog.open();
+		debugDialog.create(launchConfiguration, PROJECT_NAME);
```
Comment (jkopriva) [inline itests/org.jboss.tools.quarkus.integration.tests/src/org/jboss/tools/quarkus/integration/tests/project/RunProjectWithDebugTest.java:86]: Please also change the order of commands

File: itests/org.jboss.tools.quarkus.integration.tests/src/org/jboss/tools/quarkus/integration/tests/launch/configuration/AbstractLaunchConfigurationTest.java [hunk 6]
```diff
@@ -37,62 +34,50 @@ public static void createNewQuarkusProject(String projectName, String projectTyp
...
-
-		lc.setName(projectName + TextLabels.CONFIGURATION);
-
-		new PushButton("Browse...").click();
-		new DefaultTableItem(projectName).select();
-		new OkButton().click();
-
-		lc.apply();
-		new PushButton(TextLabels.CLOSE).click();
+	public void createNewQuarkusConfiguration(String projectName, RunConfigurationsDialog runDialog,
```
Comment (jkopriva) [inline itests/org.jboss.tools.quarkus.integration.tests/src/org/jboss/tools/quarkus/integration/tests/launch/configuration/AbstractLaunchConfigurationTest.java:45]: you can change this to return QuarkusLaunchConfigurationTabGroup not void, so you will need to call constructor in this method.

**Review by jkopriva — CHANGES_REQUESTED** [review]

**Review by jkopriva — APPROVED** [review]

**Pull request merged**